consensus/src/checkpoint.rs

```diff
@@ -74,8 +74,13 @@ pub fn is_short_range(
 #[cfg(test)]
 mod tests {
```

Comment:
Bug: While going through the is_short_range function from the spec:

```rust
fn isShortRange(C1, C2) -> bool
{
    // Get consensus state from top blocks of each chain
    let S1 = cState(C1);
    let S2 = cState(C2);

    let check = | S1, S2 | {
      if S1.epoch_count == S2.epoch_count + 1  && epochSlot(S2) >= 2/3*slots_per_epoch {
        // S1 is one epoch ahead of S2 and S2 is not in the seed update range
        return S1.staking_epoch_data.lock_checkpoint == S2.next_epoch_data.lock_checkpoint
      }
      else {
        return false
      }
    };

    if S1.epoch_count == S2.epoch_count {
        // Simple case: blocks have same previous epoch, so compare previous epochs' lock_checkpoints
        return S1.staking_epoch_data.lock_checkpoint == S2.staking_epoch_data.lock_checkpoint
    }
    else {
        // Check for previous epoch case using both orientations
        return check(S1, S2) || check(S2, S1)
    }
}
```
it appears the impl here is missing the both way check:
`return check(S1, S2) || check(S2, S1)`

we should be checking it both ways

Reply:
I've implemented `return check(S1, S2) || check(S2, S1)` recursively but encountered an unreachable expression. And encounter stack overflows as the following. That's why I made this change:
![Screenshot (271)](https://user-images.githubusercontent.com/8589224/148172293-75316ed8-6987-4ac7-a17d-4c5ae26e326f.png)

Reply:
@creativcoder thank you for patiently waiting. Resolved.